chemical_composition/chemical_composition.py

```
@@ -161,10 +159,95 @@ def clear(self):
         self.unimod_at_pos.clear()
 
         self.peptide = None
+        self.modifications = None
+        self.glycan = None
         self.addon = None
         for k in list(self.keys()):
             del self[k]
 
+    def use(
+        self,
+        sequence=None,
+        formula=None,
+        modifications=None,
+        glycan=None,
+        deprecated_format=None,
+    ):
+        """
+        Re-initialize the class with a new set of sequence/formula/modifications/glycan
+
+        This is helpful if one wants to use the same class instance
+        for multiple molecules since it removes class instantiation overhead.
+
+        Args:
+            sequence (str): peptide sequence that should be used for the instance, e.g. "PEPTIDE"
+            formula (str): chemical formula(s) that will be added/subtracted from the instance, e.g. "-H2O"
+            modifications (str): modifications of the peptide sequence, given as "Unimod:pos", e.g. "Oxidation:2"
+            glycan (str): glycan composition that will be added to the instance, e.g. "HexNAc(2)Hex(5)"
+            deprecated_format (str): old input format, using either "PEPTIDE+Formula-Formula" or "PEPTIDE#Unimod:pos"
+
+        Note:
+            Will clear the current chemical composition dict!
+        """
+
+        self.clear()
+        if sequence is not None:
+            self.add_amino_acids(sequence)
+            self["O"] += 1
+            self["H"] += 2
+
+        if modifications is not None:
+            self.add_modifications(modifications)
+
+        if glycan is not None:
+            self.add_glycan(glycan)
+
+        if formula is not None:
+            self.parse_chemical_formula(formula)
+
+        if deprecated_format is not None:
+            if (
+                sequence is not None
+                or formula is not None
+                or modifications is not None
+                or glycan is not None
+            ):
+                sys.exit(
```

I'd always prefer to raise an Exception instead of sys.exit

---

Fair, I'll change it. That goes for all sys.exits in here, right?

---

Yes, I'd prefer to use Exceptions, especially in a library, in a CLI program there are some good reasons.
Thing is, you can't catch a `sys.exit()` with `execept Exception:`, only with an `except:`